azuread/configuration.go

```diff
@@ -188,8 +205,15 @@ func (p *azureFedAuthConfig) provideActiveDirectoryToken(ctx context.Context, se
 			cred, err = azidentity.NewManagedIdentityCredential(nil)
 		}
 	case ActiveDirectoryInteractive:
-		cred, err = azidentity.NewInteractiveBrowserCredential(&azidentity.InteractiveBrowserCredentialOptions{AuthorityHost: azidentity.AuthorityHost(authority), ClientID: p.applicationClientID})
+		cloudConfig, cerr := getCloudConfig(authority)

```

Comment:
why do we need to look up the cloud config? There are TDS frontends that support AAD auth whose authority URL is not one of the standard Azure URLs.

Reply:
Breaking change in az identity that I mentioned.  I would doubt we have many people using the interactive option

Reply:
We could alternatively support them passing the full configuration

Reply:
Either way — this is blocked unless we update to use go1.18 I think

Reply:
sqlcmd uses interactive

Reply:
hah, fair enough

Reply:
this seems to work

```go
	case ActiveDirectoryInteractive:
		c := cloud.Configuration{ActiveDirectoryAuthorityHost: authority}
		config := azcore.ClientOptions{Cloud: c}
		cred, err = azidentity.NewInteractiveBrowserCredential(&azidentity.InteractiveBrowserCredentialOptions{ClientOptions: config, ClientID: p.applicationClientID})

```

Reply:
that's effectively what's in the PR now, with yours having more done in-line

		cloudConfig := cloud.Configuration{ActiveDirectoryAuthorityHost: authority, Services: map[cloud.ServiceName]cloud.ServiceConfiguration{}}
		opts := &azidentity.InteractiveBrowserCredentialOptions{ClientID: p.applicationClientID}
		opts.Cloud = cloudConfig
		cred, err = azidentity.NewInteractiveBrowserCredential(opts)